### PR TITLE
Enhancment/add error for summary

### PR DESCRIPTION
**Pull request by volovikariel**
Aims to address #96.

There's a few parts that I'm unsure about, that's why this is starting off as a draft PR. 

I recommend looking through the changes commit by commit - the diff (on github) between the current state of the branch and the main branch is very hard to follow.

### CODE REVIEW

File: cmd/summary.go [hunk 1]
```diff
@@ -233,17 +155,91 @@ func GetSummaryCommand() *cobra.Command {
...
 			}
-			pterm.Error.Println("wrong number of arguments, expecting two (2)")
 			return nil
 		},
 	}
 	cmd.Flags().BoolP("no-color", "n", false, "Disable color and style output (very useful for CI/CD)")
 	return cmd
 }
 
+func listenForUpdates(updateChan chan *model.ProgressUpdate, errorChan chan model.ProgressError, doneChan chan bool, failed bool) {
```
Comment (volovikariel) [inline cmd/summary.go:171]: The `failed` bool is never updated.

Currently - even if there's a `spinner.Fail` call, we mark  it as done and return nothing. No errors or anything like that. Is this desired behaviour?

Should a failure in one `listenForUpdates` call result in a failure in all other ones?
If so - should we use a `failed *bool` instead of `failed bool`, and set it to true wherever there's a `spinner.Fail` call?

Or should we instead utilize the `errorChan`? What is the difference between `failed` and `errorChan`?

File: cmd/summary.go [hunk 2]
```diff
@@ -161,14 +83,14 @@ func GetSummaryCommand() *cobra.Command {
 							return er
 						}
 						return nil
+					} else {
+						return errors.New(fmt.Sprintf("Host %q is not supported.\nSupported hosts are: %v", specUrl.Host, supportedHosts))
```
Comment (volovikariel) [inline cmd/summary.go:87]: Should we just return an error, or should I also do a `pterm.Error.println(...)`?

File: cmd/summary.go [hunk 3]
```diff
@@ -161,14 +83,14 @@ func GetSummaryCommand() *cobra.Command {
...
 					}
 
 				} else {
-					pterm.Error.Println("Two arguments are required to compare left and right OpenAPI Specifications.")
-					return nil
+					return errors.New(fmt.Sprintf("Error parsing url: %v", err))
```
Comment (volovikariel) [inline cmd/summary.go:91]: Should we just return an error, or should I also do a `pterm.Error.println(...)`?

File: cmd/summary.go [hunk 4]
```diff
@@ -233,17 +155,91 @@ func GetSummaryCommand() *cobra.Command {
 						}
 						return errors.Join(errs...)
 					}
-					return nil
 				}
+			} else {
+				// if an invalid number of arguments are provided, print usage
+				PrintHowToUse("summary")
+				return errors.New("invalid number of arguments provided")
```
Comment (volovikariel) [inline cmd/summary.go:162]: Should we just return an error, or should I also do a `pterm.Error.println(...)`?

**Comment by daveshanley** [comment]
These changes were mostly implemented during the development of `v0.0.56`

Thank you for your work on this however.
